### PR TITLE
wip 0.3.0

### PR DESCRIPTION
**Pull request by chrisguida**

### CODE REVIEW

File: manifest.yaml [hunk 1]
```diff
@@ -1,12 +1,79 @@
 compat: v0
 id: c-lightning
-version: 0.10.0
+version: 0.10.1
```
Comment (elvece) [inline manifest.yaml:3]: keep the native semver - just update with our internal last digit ie. 0.10.0.1

File: Dockerfile [hunk 2]
```diff
@@ -79,10 +79,12 @@ RUN test -n "$BITCOIN_VERSION"
...
 COPY --from=builder /usr/local /usr/local
 COPY --from=bitcoin-core /bitcoin-${BITCOIN_VERSION}/src/bitcoin-cli /usr/local/bin
-ADD ./c-lightning-http-plugin/target/armv7-unknown-linux-musleabihf/release/c-lightning-http-plugin /usr/local/libexec/c-lightning/plugins/c-lightning-http-plugin
-ADD ./configurator/target/armv7-unknown-linux-musleabihf/release/configurator /usr/local/bin/configurator
+ADD ./c-lightning-http-plugin/target/aarch64-unknown-linux-musl/release/c-lightning-http-plugin /usr/local/libexec/c-lightning/plugins/c-lightning-http-plugin
+ADD ./configurator/target/aarch64-unknown-linux-musl/release/configurator /usr/local/bin/configurator
 ADD ./docker_entrypoint.sh /usr/local/bin/docker_entrypoint.sh
 RUN chmod a+x /usr/local/bin/docker_entrypoint.sh
+ADD ./config.sh /usr/local/bin/config
+RUN chmod a+x /usr/local/bin/config
```
Comment (elvece) [inline Dockerfile:87]: these steps can be removed in favor of using the `config` command from the `compat` image

File: manifest.yaml [hunk 3]
```diff
@@ -1,12 +1,79 @@
...
+    system: true
+    entrypoint: config
+    args:
+      - "set"
+      - "/root"
+    mounts: {}
+    io-format: yaml
+volumes: {}
+interfaces:
+  main:
```
Comment (elvece) [inline manifest.yaml:50]: add keys:  `name` `description` to each interface (just "main" in this case)

File: manifest.yaml [hunk 4]
```diff
@@ -20,83 +87,92 @@ shm-size-mb: 64
...
-os-version-required: "^0.2.10"
-os-version-recommended: "^0.2.10"
-assets: []
 hidden-service-version: v3
 dependencies:
   btc-rpc-proxy:
     version: ">=0.2.4 <0.4.0"
     optional: Can alternatively configure an external bitcoin node.
     description: Used to fetch validated blocks.
+    critical: false
```
Comment (elvece) [inline manifest.yaml:96]: add key `recommended` - new necessary key for each dependency

File: config.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#!/bin/bash -e
```
Comment (elvece) [inline config.sh:1]: this file can be deleted in favor of using commands in `compat` image

File: Makefile [hunk 6]
```diff
@@ -11,20 +11,19 @@ CONFIGURATOR_SRC := $(shell find ./configurator/src) configurator/Cargo.toml con
 all: c-lightning.s9pk
 
 install: c-lightning.s9pk
-	appmgr install c-lightning.s9pk
+	embassy-sdk pack
```
Comment (elvece) [inline Makefile:14]: the install command can be deleted. instead, update the `bitcoind.s9pk` step below, like [here](https://github.com/Start9Labs/cups-wrapper/pull/1/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R19)

there needs to be both a `embassy-sdk pack` and `embassy-sdk verify` step

File: manifest.yaml (hunk 1 above)
Comment (chrisguida) [reply manifest.yaml:3]: This update would potentially pull in 0.10.1, which is the latest release tag from upstream

File: manifest.yaml [hunk 7]
```diff
@@ -1,115 +1,180 @@
...
+    type: docker
+    # default backup process is duplicity - EOS will have access to this image
+    image: duplicity
+    system: true
+    # command to run the backup executable, in this case, duplicity
+    entrypoint: duplicity
+    # arguments to pass into the entrypoint ie. duplicity in this case
+    # thus, the full commamnd run will be: `duplicity /root file:///mnt/backup`
+    args:
+      - /root
```
Comment (dr-bonez) [inline manifest.yaml:111]:
```suggestion
      - /root/.lightning
```

File: manifest.yaml [hunk 8]
```diff
@@ -1,115 +1,180 @@
...
+      BACKUP: "/mnt/backup"
+      main: "/root/.lightning"
+  restore:
+    type: docker
+    image: duplicity
+    system: true
+    entrypoint: duplicity
+    args:
+      - file:///mnt/backup
+      - /root
```
Comment (dr-bonez) [inline manifest.yaml:126]:
```suggestion
      - /root/.lightning
```

**Review by dr-bonez — CHANGES_REQUESTED** [review]

File: manifest.yaml [hunk 9]
```diff
@@ -1,115 +1,180 @@
...
+  instructions: instructions.md
+  docker-images: image.tar
+  icon-type: png
+main:
+  type: docker
+  image: main
+  entrypoint: "docker_entrypoint.sh"
+  args: []
+  mounts:
+    btc-rpc-proxy: "/mnt/btc-rpc-proxy"
```
[the remark this answers is not in the document]
Comment (chrisguida) [reply manifest.yaml:30]: Not sure, copied from LND: https://github.com/Start9Labs/lnd-wrapper/pull/24/files#diff-d865b06b497efc8c56c63e5bedfc86f6da6a005cdb7d30e3702286f3d918aaa6R30

File: manifest.yaml [hunk 10]
```diff
@@ -1,115 +1,180 @@
...
+  # this is the image where files from the project asset directory will go
+  compat:
+    type: assets
+  # this is a pointer volume, where the image is specified in `<pointer-id>.volume-id` and the mount point is specificed in `main.mounts.<pointer-id>`
+  btc-rpc-proxy:
+    type: pointer
+    package-id: btc-rpc-proxy
+    volume-id: main
+    path: "/root"
+    readonly: false
```
[the remark this answers is not in the document]
Comment (chrisguida) [reply manifest.yaml:72]: Copied from LND

File: manifest.yaml [hunk 11]
```diff
@@ -1,115 +1,180 @@
...
+  # TODO - update configurator to not gen cert
+  certificates:
+    type: certificate
+    # id of the volume you want the cert for
+    interface-id: main
```
[the remark this answers is not in the document]
Comment (chrisguida) [reply manifest.yaml:77]: Copied from LND. Wasn't sure about this one

File: manifest.yaml (hunk 1 above)
Comment (elvece) [reply manifest.yaml:3]: oh gotcha I did not realize - then 0.10.1 is righteous

File: manifest.yaml (hunk 11 above)
[the remark this answers is not in the document]
Comment (elvece) [reply manifest.yaml:77]: the `certificates` key is needed only if the service requires a cert - for LND this is true for the grpc and rest interfaces. I dont think this is something c-lightning requires.

File: manifest.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (elvece) [reply manifest.yaml:30]: @dr-bonez don't you need a mount for pointers?

File: manifest.yaml (hunk 10 above)
[the remark this answers is not in the document]
Comment (elvece) [reply manifest.yaml:72]: @dr-bonez isn't this needed in order to configure proxy for c-lightning?

File: manifest.yaml (hunk 9 above)
[the remark this answers is not in the document]
Comment (dr-bonez) [reply manifest.yaml:30]: Not for config pointers. This is for filesystem access only

File: manifest.yaml (hunk 10 above)
[the remark this answers is not in the document]
Comment (dr-bonez) [reply manifest.yaml:72]: No

File: manifest.yaml (hunk 11 above)
[the remark this answers is not in the document]
Comment (chrisguida) [reply manifest.yaml:77]: I wasn't sure, but I think you're right that there's no SSL in c-lightning. Removing.

File: manifest.yaml (hunk 10 above)
[the remark this answers is not in the document]
Comment (chrisguida) [reply manifest.yaml:72]: Removed

**Pull request closed without merging**